### PR TITLE
CNX-9292 apply unit of work for auto cad

### PR DESCRIPTION
**Pull request by oguzhankoral**
UoWs added for receive and send. And UoWs in builders replaced with `ISpeckleConverterToSpeckle _converter` and `ISpeckleConverterToHost _converter` dependency.

### CODE REVIEW

File: DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/DependencyInjection/AutofacAutocadModule.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ protected override void Load(ContainerBuilder builder)
     // Operations
     builder.RegisterType<SendOperation<(DBObject obj, string applicationId)>>().InstancePerLifetimeScope();
     builder.RegisterType<ReceiveOperation>().SingleInstance();
-    builder.RegisterType<SyncToUIThread>().As<ISyncToMainThread>().SingleInstance();
+    builder.RegisterType<SyncToUIThread>().As<ISyncToMainThread>().SingleInstance().AutoActivate(); // Auto activation makes SynchronizationContext available right after building. Otherwise we were missing it in UoWs.
```
Comment (AlanRynne) [inline DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/DependencyInjection/AutofacAutocadModule.cs:53]: Does this need propagating to Rhino/Revit?

**Review by AlanRynne — APPROVED** [review]
Just one question regarding the AutoActivate 🙌🏼

File: DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/DependencyInjection/AutofacAutocadModule.cs (hunk 1 above)
Comment (oguzhankoral) [reply DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/DependencyInjection/AutofacAutocadModule.cs:53]: I believe it is more of a MainThread thing, Rhino should be fine without AutoActive, but not sure about Revit. And @connorivy was saying he did something Revit specific? If `ISyncToMainThread` interface work with Revit too, we should align it I believe.

Comment (AlanRynne) [reply DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/DependencyInjection/AutofacAutocadModule.cs:53]: Ok no blocker for this then! In we go!

**Pull request merged**